src/materialized/src/version_check.rs

```diff
@@ -50,34 +53,66 @@ pub async fn check_version_loop(telemetry_url: String, cluster_id: String) {
         Ok(_) => (),
         Err(e) => debug!("unable to parse fetched latest version: {}", e),
     }
+
+    loop {
+        tokio_sleep(TokioDuration::from_secs(TELEMETRY_FREQUENCY_SECS)).await;
+
+        fetch_latest_version(&version_url, start_time).await;
+    }
 }
 
-async fn fetch_latest_version(
-    telemetry_url: &str,
-    cluster_id: &str,
-    current_version: &str,
-) -> anyhow::Result<String> {
-    let version_url = format!("{}/api/v1/version/{}", telemetry_url, cluster_id);
-    let version_request = V1VersionRequest {
-        version: current_version.to_string(),
-    };
-
-    let resp = reqwest::Client::new()
-        .post(&version_url)
-        .timeout(Duration::from_secs(10))
-        .json(&version_request)
-        .send()
-        .await?;
-    if !resp.status().is_success() {
-        bail!("failed request: {}", resp.status());
+async fn fetch_latest_version(telemetry_url: &str, start_time: Instant) -> String {
+    RetryBuilder::new()
+        .max_sleep(None)
```

Comment:
What is the maximum sleep that can be realistically achieved here? It seems to me that once a day should be the longest to go between checking for versions?

Reply:
good catch, this was potentially infinite, the new commit max's it out at once an hour.